lib/IRGen/GenCall.cpp

```diff
@@ -938,13 +938,16 @@ llvm::Type *SignatureExpansion::expandExternalSignatureTypes() {
         break;
       }
 
-      // If the coercion type is a struct, we need to expand it.
-      auto type = AI.getCoerceToType();
-      if (auto expandedType = dyn_cast<llvm::StructType>(type)) {
-        for (size_t j = 0, e = expandedType->getNumElements(); j != e; ++j)
-          ParamIRTypes.push_back(expandedType->getElementType(j));
+      // If the coercion type is a struct which can be flattened, we need to
+      // expand it.
+      auto *ArgTy = AI.getCoerceToType();
```

Comment:
'type' was fine as the name of this variable, and this code would probably look somewhat cleaner if you just dyn_cast'ed to StructType outside of the condition.

Reply:
I was trying to use similar names in the various places.  Renamed it to `coercedTy`.